File: 01. Lenguajes de proposito general/03. Persistencia/01. Directorios/C++/Directorios.cpp

#include <sys/stat.h>	// mkdir
#include <stdio.h>		// rename, remove

// Estctura para verificar si un directorio existe
struct stat st = {0};

main () {

	// Direcciones
	char* direccion = 'C:\\carpeta\\carpeta2\\carpetaN\\archivo.extensión';	// Direccion absoluta (Desde el la unidad de disco duro)
	char* direccion = './carpeta1/carpeta2/carpetaN/archivo.extensión';		// Direccion relativa (Desde el directorio actual) (Los dos puntos (..) significa obtener carpeta padre a la actual)

	// Acceso
	direccion 					// Obtener dirección especficada
	direccion + "/.."			// Obtener padre
	direccion + "/nombre"		// Obtener hijo

	// Creación
	mkdir(direccion, 777);					// Crear direcctorio		
	fclose(fopen(direccion, "w"));			// Crear archivo

	// Modificación
	rename(direccionActual, direccionNueva);	// Renombrar direcctorio		

	// Eliminación
	remove(direccion);			// Remover archivo
	NO IMPLEMENTADO				// Remover directorio

	// Infromación
	stat(direccion, &st) == 0 && S_ISDIR(st.st_mode)							// Verificar si es la ruta es un direcctorio
	stat(direccion, &st) == 0 && S_ISREG(st.st_mode)							// Verificar si es la ruta es un archivo
	stat(direccion, &st) == 0 && (S_ISREG(st.st_mode) && S_ISREG(st.st_mode))	// Verificar si la ruta existe

}

File: 01. Lenguajes de proposito general/03. Persistencia/01. Directorios/Python/Directorios.py

# Modulo para el manejo de archivos
import os

# Direcciones
direccion = r'C:\carpeta1\carpeta2\carpetaN\archivo.extensión'	# Direccion absoluta (Desde el la unidad de disco duro)
direccion = 'carpeta1/carpeta1/carpetaN/archivo.extensión'		# Direccion relativa (Desde la raiz del proyecto)

# Acceso
direccion 										# Obtener dirección especficada
os.path.dirname(os.path.realpath(direccion))	# Obtener padre
direccion + r"\nombre"							# Obtener hijo (Direccion absoluta)
direccion + "/nombre"							# Obtener hijo (Direccion relativa)

# Creación
os.mkdir(direccion)				# Crear direcctorio		
open(direccion, 'w').close()	# Crear archivo

# Modificación
os.rename(direccionActual, direccionNuena)		# Renombrear direcctorio

# Eliminación
os.remove(direccion)					# Remover archivo
os.rmdir(direccion)						# Remover directorio vacio
import shutil; shutil.rmtree(direccion)	# Remover directorio con el contenido

# Apuntador
os.chdir(direccion)						# Aceder al directorio especificado
os.walk("nombre") 						# Aceder al directorio hijo por nombre
os.chdir(os.path.dirname(os.getcwd()))	# Aceder al directorio padre

# Infromación
os.path.isdir(direccion)	# Verificar si es la ruta es un direcctorio
os.path.isfile(direccion)	# Verificar si es la ruta es un archivo
os.path.exists(direccion)	# Verificar si la ruta existe
os.path.basename(direccion) # Obtener nombre del directorio o archivo
os.getcwd()     			# Obtener directorio actual como String
os.getcwdb()    			# Obtener directorio actual como objeto
os.listdir()				# Listar nombres de directorios y archivos contenidos en el directorio actual
os.listdir(direccion)		# Listar nombres de directorios y archivos contenidos en el directorio especificado


